### PR TITLE
Datasource remote operations

### LINKED ISSUE #1601: Integrate igTimePicker into the igGrid
Modify the datasource to support time columns. Update filtering algorithm to support time columns.

### PR DESCRIPTION
**Pull request by bkulov**
Closes #1601 

Encode the type of the column in the url (in case of remote operations). This will allow the server to properly identify the column type. Required for time columns as they use date objects.

### CODE REVIEW

File: src/js/modules/infragistics.datasource.js [hunk 1]
```diff
@@ -5269,12 +5285,26 @@
...
+						}
+
+						params.summariesParams[ key ] = methodsStr.slice(0, -1);
 					}
 				}
 			}
 		},
+		_getSchemaField: function (fieldName) {
+			if (this.settings.schema && this.settings.schema.fields.find) {
+				return this.settings.schema.fields.find(function (f) { return f.name === fieldName; });
```
Comment (mpavlinov) [inline src/js/modules/infragistics.datasource.js]: @bkulov : Array.prototype.find is not available in Internet Explorer. This code will throw an error there. I'm not sure we provide a polyfill for it. Can you check it please.

Comment (bkulov) [reply src/js/modules/infragistics.datasource.js]: Sure. Will modify it.

Comment (mpavlinov) [reply src/js/modules/infragistics.datasource.js]: @bkulov Agree. It won't throw and error.

Comment (bkulov) [reply src/js/modules/infragistics.datasource.js]: Done :)

**Pull request merged**